pandas/_testing.py

```diff
@@ -485,23 +485,37 @@ def ensure_clean(filename=None, return_filelike=False):
     return_filelike : bool (default False)
         if True, returns a file-like which is *always* cleaned. Necessary for
         savefig and other functions which want to append extensions.
```

Comment:
why don’t list encoding , mode here explicitly

Reply:
and in the signature

Reply:
`encoding` and `mode` are not accepted in `mkstemp`.

The only shared parameters are `prefix`, `suffix` (corresponds to `filename` in our signature), and `dir`, though uncertain that `prefix` and `dir` are important to make explicit.

Reply:
Looking back on comment in OP is the main thing needed for now just encoding? If so I think better to just list explicitly and not pass to mkstemp

Reply:
> If so I think better to just list explicitly and not pass to mkstemp

The whole point is to use the `**kwargs` in `ensure_clean` as a pass through to `mkstemp` or `TemporaryFile`, depending on whether `return_filelike` is true or false.

Explicitly listing arguments that apply to one but not the other is going to be confusing.